src/main/main.cpp

```diff
@@ -3,29 +3,71 @@
  * Main entrypoint for the tvp executable
  */
 
+#include "debugger/debugger.h"
 #include "gameboy/gameboy.h"
 
+#include <cxxopts.hpp>
+
+using namespace std;
+using namespace cpu;
+using namespace gpu;
+using namespace video;
+using namespace memory;
+using namespace gameboy;
+using namespace debugger;
+using namespace cartridge;
+using namespace controller;
+
 int main(int argc, char *argv[]) {
 	ios_base::sync_with_stdio(false);
 
-	auto rom_path = string("");
-	if (argc < 2) {
-		cerr << "Please provide the name of a ROM file or press q/Q to exit"
-		     << endl;
-		getline(cin, rom_path);
-		if (rom_path == "q" || rom_path == "Q") {
-			cout << "Exiting TVP" << endl;
-			exit(0);
-		}
-	} else {
-		rom_path = string(argv[1]);
+	auto cmdline_args_parser =
+	    cxxopts::Options("tvp", "Welcome to tvp - The GameBoy Emulator!");
+
+	// clang-format off
+	// TODO: ClangFormat for this style is bad, fix rules
+	cmdline_args_parser.add_options()
+		("r,rom", "Path to a GameBoy ROM file - REQUIRED",
+			cxxopts::value<string>())
+		("d,debug", "Enable the debugger",
+			cxxopts::value<bool>()->default_value("false"))
+		("h,help", "Print this information");
+	// clang-format on
+
+	auto parsed_args = cmdline_args_parser.parse(argc, argv);
+
+	// Print help info if -h is passed
+	auto help = parsed_args["help"].as<bool>();
+	if (help) {
+		cout << cmdline_args_parser.help();
+		exit(0);
+	}
+
+	// Get ROM path and fail with help message if arg not parsed
+	string rom_path = "";
+	try {
+		rom_path = parsed_args["rom"].as<string>();
+	} catch (exception &e) {
+		cout << cmdline_args_parser.help();
```

Comment:
Pretty Cool. This will make the entire thing so much more clean.

Reply:
Yep, this library cxxopts is quite nice.